File: RubyCivitas/lib/sorpresa_porcasahotel.rb
module Civitas
  class Sorpresa_porcasahotel < Sorpresa
    
    def initialize(valor, texto)
      
      super(texto)
      @valor = valor
      
    end
    
    def aplicar_a_jugador(actual, todos)

      if jugador_correcto(actual, todos)

        informe(actual, todos)
        todos[actual].modificar_saldo(@valor*(todos[actual].cantidad_casas_hoteles))

      end
    end
  
  end
end

File: civitasjava/src/Civitas/SorpresaPorCasaHotel.java
package civitas;

import java.util.ArrayList;


public class SorpresaPorCasaHotel extends Sorpresa{
    private int valor;
    
    SorpresaPorCasaHotel(int valor, String texto){
        super(texto);
        this.valor = valor;
    }
    
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            todos.get(actual).modificarSaldo(valor * 
                    todos.get(actual).cantidadCasasHoteles());
        }
    }
    
    @Override
    public String toString(){
        return "SorpresaPorCasaHotel{ Valor: " +valor +super.toString() + "}";
    }
}
